### PR TITLE
Fix ZHA device reconfiguration dialog

### LINKED ISSUE #10993: ZHA Device Reconfiguration Dialog only opens once
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Zigbee devices Reconfigure Dialog does not open again once closed

When on a zigbee device (driven by ZHA), I click on "RECONFIGURE DEVICE" it only works the first time.
Once I close the dialog it stops working and I've to reload the whole webapp to get it working again

I'm seeing this in the home assistant logs:

```
2021-12-21 18:33:31 ERROR (MainThread) [frontend.js.latest.202112150] https://zuniga.trevi.me:8123/frontend_latest
/17dc2469.js:246:484 TypeError: this._clusterConfigurationStatuses is undefined
```

### Describe the behavior you expected

The dialog opens

### Steps to reproduce the issue

1. Open the details of a ZHA device
2. Click on Reconfigure Device
3. Close the dialog
4. Click again

The dialog won't open ant errors will be shown both on logs and JS console

### What version of Home Assistant Core has the issue?

2021.12.4

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox 95 but also the Android WebApp

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

```txt
Uncaught (in promise) TypeError: this._clusterConfigurationStatuses is undefined
    value dialog-zha-reconfigure-device.ts:228
    update app.2b82fe2c.js:1243
    performUpdate app.2b82fe2c.js:1243
    scheduleUpdate app.2b82fe2c.js:1243
    _$EC app.2b82fe2c.js:1243
    requestUpdate app.2b82fe2c.js:1243
    set app.2b82fe2c.js:1243
    value app.2b82fe2c.js:1243
    value app.2b82fe2c.js:1243
    value home-assistant.ts:75
    _$AE app.2b82fe2c.js:1243
    performUpdate app.2b82fe2c.js:1243
    scheduleUpdate app.2b82fe2c.js:1243
    _$EC app.2b82fe2c.js:1243
    requestUpdate app.2b82fe2c.js:1243
    set app.2b82fe2c.js:1243
    value app.2b82fe2c.js:1243
    hassConnected app.2b82fe2c.js:1243
    s core.47513876.js:1
    l core.47513876.js:1
    l core.47513876.js:1
    _handleMessage core.47513876.js:1
dialog-zha-reconfigure-device.ts:228
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dmulcahey**
## Proposed change
This PR corrects an error with the device reconfiguration dialog for ZHA. 

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #10993 fixes https://github.com/home-assistant/core/issues/55896
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 1]
```diff
@@ -225,10 +226,11 @@ class DialogZHAReconfigureDevice extends LitElement {
                   )}
                 </h2>
 
-                ${this._clusterConfigurationStatuses!.size > 0
+                ${this._clusterConfigurationStatuses &&
+                this._clusterConfigurationStatuses.size > 0
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:230]:
```suggestion
                ${this._clusterConfigurationStatuses?.size
```

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts (hunk 1 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts]: I had that first... something is up with my dev environment. It still flags that with the "object possibly null or undefined" when I use `?.`

Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts]: actually it breaks CI too...

Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts]: never mind... needed more coffee

**Review by bramkragten — APPROVED** [review]

**Pull request merged**